src/Native/Windows/CMakeLists.txt

```diff
@@ -1,23 +1,123 @@
+
+# Statically linked CRT (libcmt[d].lib, libvcruntime[d].lib and libucrt[d].lib) by default. This is done to avoid
+# linking in VCRUNTIME140.DLL for a simplified xcopy experience by reducing the dependency on VC REDIST.
+#
+# For Release builds, we shall dynamically link into uCRT [ucrtbase.dll] (which is pushed down as a Windows Update on downlevel OS) but
+# wont do the same for debug/checked builds since ucrtbased.dll is not redistributable and Debug/Checked builds are not
+# production-time scenarios.
+if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL "RELEASE")
+    add_compile_options(/MT)
+    add_compile_options(/GL)
+    add_compile_options(/O1)
+    set(__LinkLibraries "${__LinkLibraries} libcmt.lib")
+    set(__LinkLibraries "${__LinkLibraries} libvcruntime.lib")
+    set(STATIC_UCRT_LIB "libucrt.lib")
+    set(DYNAMIC_UCRT_LIB "ucrt.lib")
+else()
+    add_compile_options(/MTd)
+    set(__LinkLibraries "${__LinkLibraries} libcmtd.lib")
+    set(__LinkLibraries "${__LinkLibraries} libvcruntimed.lib")
+    set(STATIC_UCRT_LIB "libucrtd.lib")
+    set(DYNAMIC_UCRT_LIB "ucrtd.lib")
+endif()
+
+# Linker flags
+set(__LinkArgs "${__LinkArgs} /INCREMENTAL:NO")
+set(__LinkArgs "${__LinkArgs} /MANIFEST:NO")            #Do not create Side-by-Side Assembly Manifest
+set(__LinkArgs "${__LinkArgs} /SUBSYSTEM:WINDOWS,6.00") #windows subsystem
+set(__LinkArgs "${__LinkArgs} /LARGEADDRESSAWARE")      # can handle addresses larger than 2 gigabytes
+set(__LinkArgs "${__LinkArgs} /RELEASE")                #sets the checksum in the header
+set(__LinkArgs "${__LinkArgs} /NXCOMPAT")               #Compatible with Data Execution Prevention
+set(__LinkArgs "${__LinkArgs} /DYNAMICBASE")            #Use address space layout randomization
+set(__LinkArgs "${__LinkArgs} /DEBUGTYPE:cv,fixup")     #debugging format
+set(__LinkArgs "${__LinkArgs} /PDBCOMPRESS")            #shrink pdb size
+set(__LinkArgs "${__LinkArgs} /DEBUG")
+set(__LinkArgs "${__LinkArgs} /IGNORE:4197,4013,4254,4070,4221")
+
+if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL "RELEASE")
+    # Release build specific flags
+    set(__LinkArgs "${__LinkArgs} /LTCG /OPT:REF /OPT:ICF /INCREMENTAL:NO")
+    
+    # Force uCRT to be dynamically linked for Release build
+    set(__LinkArgs "${__LinkArgs} /NODEFAULTLIB:libucrt.lib /DEFAULTLIB:ucrt.lib")
+    
+    if (CMAKE_SYSTEM_PROCESSOR STREQUAL x86)
+        set(__LinkArgs "${__LinkArgs} /SAFESEH")
+    endif()
+else()
+    # Debug build specific flags
+    set(__LinkArgs "/NOVCFEATURE ${__LinkArgs}")
+endif()
+
+# Temporary until cmake has VS generators for arm64
+if("${__BuildArch}" STREQUAL "arm64")
+    set(__LinkArgs "${__LinkArgs} /machine:arm64")
+endif()
+
+if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "amd64")
     add_definitions(-DBIT64=1)
-elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l)
+elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL "armv7l")
```

Comment:
It's possible. Most of this is taken from the native coreclr build which is used cross-platform and at the very least the comment about clang is wrong since we're not using clang.

What about Windows ARM though? Some searching suggests CMAKE_SYSTEM_PROCESSOR is set to %PROCESSOR_ARCHITECTURE% which might be set to "arm" but won't be ever set to "arm7l". The VS generator doesn't need the triple either way so I can remove it.